util/waiter.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import "sync"
+
```

Comment:
PTAL at https://godoc.org/golang.org/x/sync/errgroup

Reply:
IMO, It has a bit different semantic